### ISSUE TITLE
pSortableColumn and pSortIcon Accessibility Issues

### ISSUE THREAD

**Issue by aciesla4** [primefaces/primeng#18601]
### Describe the bug

The pSortableColumn and pSortIcon have accessibility violations when running the IBM Equal Accessibility Checker. The pSortableColumn has the issues: Table structure elements cannot specify an explicit 'role' within table containers and A tabbable element must have a valid widget role. The pSortIcon has the issue: A non-decorative SVG element must have an accessible name.

### Reason for not contributing a PR

- [x] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/pgewopns-2k6snkqx

### Environment

Stackblitz 

### Angular version

20.0.4

### PrimeNG version

v20

### Steps to reproduce the behavior

1. Install the [IBM Equal Accessibility Checker ](https://chromewebstore.google.com/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp)
2. Open the stackblitz 
3. Run the accessibility checker 
4. Filter on violations only and note the errors: 'An explicit ARIA 'role' is not valid for <th> element within an ARIA role 'table' per the ARIA in HTML specification', 'The tabbable element's role 'rowheader' is not a widget role', and 'The SVG element has no accessible name'

### Expected behavior

No accessibility violations on the pSortableColumn are present. The pSortIcon provides a parameter to provide an aria-label.
